packages/vite/src/node/config.ts

```diff
@@ -527,12 +527,12 @@ export function mergeConfig(
       continue
     }
 
-    // root fields that require special handling
-    if (existing != null && isRoot) {
-      if (key === 'alias') {
+    // fields that require special handling
+    if (existing != null) {
+      if (key === 'alias' && !isRoot) {
```

Comment:
This might affect nested options like `rollupOptions` unexpected. Maybe we should change the third argument to the `rootPath` and ensure only resolve `alias` under `resolve` field

Reply:
Good point - are you thinking `rootPath` would be the key if the previous iteration (i.e. `resolve`)? Should `assetsInclude` check for a missing `rootPath` (implying it's already a root iteration)? Something like:
```js
export function mergeConfig(
  a: Record<string, any>,
  b: Record<string, any>,
  rootPath?: string
): Record<string, any> {
  // ...
  for (const key in b) {
    // ...
    // fields that require special handling
    if (existing != null) {
      if (key === 'alias' && rootPath === 'resolve') {
        merged[key] = mergeAlias(existing, value)
        continue
      } else if (key === 'assetsInclude' && !rootPath) {
        merged[key] = [].concat(existing, value)
        continue
      }
    }
    // ...
  }
  return merged
}
```
This implementation passes with the new tests. Something like `prevKey` might make more sense here but I'm good either way.

Reply:
Yes. But it's a bit tricky that we would need to support boolean as well to avoid introducing breaking changes.

Reply:
Hmm, that's true. I'd personally lean towards a new minor version with a breaking change over having to support both cases, but I'll defer to you on that. I'd guess most people probably aren't using the param anyways

Reply:
@antfu just wanted to bump this - let me know how you want to proceed here

Reply:
@chrissantamaria Let's keep the public interface as-is to avoid breaking changes, and move the recursive logic into a new internal function so we can have `rootPath` as an argument. If you want, I can take over this PR to make it ready.

Reply:
Sure! Should be minimal changes so feel free to take over the PR and implement as you see fit.

Reply:
Done, thanks!